ethereum/solidity/contracts/lib/tree/namespace/TreeHasher.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+import "../Constants.sol";
+import "./NamespaceNode.sol";
+
+/// @notice Get the minimum namespace.
+// solhint-disable-next-line func-visibility
+function namespaceMin(bytes8 l, bytes8 r) pure returns (bytes8) {
+    if (uint64(l) < uint64(r)) {
+        return l;
+    } else {
+        return r;
+    }
+}
+
+/// @notice Get the maximum namespace.
+// solhint-disable-next-line func-visibility
+function namespaceMax(bytes8 l, bytes8 r) pure returns (bytes8) {
+    if (uint64(l) > uint64(r)) {
+        return l;
+    } else {
+        return r;
+    }
+}
+
+/// @notice Hash a leaf node.
+/// @param minmaxNID Namespace ID.
+/// @param data Raw data of the leaf.
+// solhint-disable-next-line func-visibility
+function leafDigest(bytes8 minmaxNID, bytes memory data) pure returns (NamespaceNode memory) {
+    bytes32 digest = sha256(abi.encodePacked(Constants.LEAF_PREFIX, minmaxNID, minmaxNID, data));
+    NamespaceNode memory node = NamespaceNode(minmaxNID, minmaxNID, digest);
+    return node;
+}
+
+/// @notice Hash an internal node.
+/// @param l Left child.
+/// @param r Right child.
+// solhint-disable-next-line func-visibility
+function nodeDigest(NamespaceNode memory l, NamespaceNode memory r) pure returns (NamespaceNode memory) {
```

Comment:
```suggestion
function nodeDigest(NamespaceNode memory l, NamespaceNode memory r) pure returns (NamespaceNode memory) {
    // More details in https://github.com/celestiaorg/celestia-specs/blob/master/src/specs/data_structures.md#namespace-merkle-tree
```

Reply:
Fixed in [7e9a48f](https://github.com/celestiaorg/quantum-gravity-bridge/pull/32/commits/7e9a48f41678e604ecb9f0f8d3bb4d7d1300a345)